### ISSUE TITLE
New feature: reindex matched results

### ISSUE THREAD

**Issue by HsunGong** [jjw24/Wox#84]
I wonder **the strategy of `matching`**. 

### Steps to reproduce
*1* Initially, Figure 1 shows in `Wox`:
![image](https://user-images.githubusercontent.com/32933986/70023176-8318a500-15d1-11ea-9299-5a2b161b96ef.png)
![image](https://user-images.githubusercontent.com/32933986/70023257-d12da880-15d1-11ea-83ac-25106c5008f4.png)
*2* Then, I set it to the top
![image](https://user-images.githubusercontent.com/32933986/70023276-e571a580-15d1-11ea-987e-823ba7d1af8f.png)
*3* but find that the results order remains unchanged if query has changed a little.
![image](https://user-images.githubusercontent.com/32933986/70023382-44cfb580-15d2-11ea-80a4-2094067b25ab.png)

*4* What I want finally is 
![image](https://user-images.githubusercontent.com/32933986/70024960-94b07b80-15d6-11ea-9933-1d7de1b6221c.png)

### Here is my thought to solve this.

The strategy of matching is the socket between `Plugins` and `Wox-main`. If the query has changed, Wox's setting of `Top` doesn't work anymore.

The improvement maybe done as follows.
Origin flow(NOT SURE): `Query -> Plugins -> matchings -> Wox reindex(If the query + result == setted ones) -> show to users`.
Redesign flow: `Query -> Plugins -> matchings -> Wox index matching(If query.contains(setted ones)) -> show to users`

**Comment by jjw24** [comment]
Yes from memory it does exact matching with the query keyword so obviously if the keyword is changed then it will not pick up the match. Also found you can only assign one query keyword to a record. So I am not too sure the original intended design was for setting record to the top. 

However if you intend to use 'exce' to bring up 'Excel: Easily…' then what would be a scenario where you type 'excel'?

**Comment by HsunGong** [comment]
When I type a long word, for example, `Windows Terminal`, I can not find a solution from the list if `te`, `ter`, `term` is typed. Then I have to type `terminal`, or set all to the top.

Other cases is that, if I reindex the program, and all `top` disappeared.

**Comment by jjw24** [comment]
> When I type a long word, for example, `Windows Terminal`, I can not find a solution from the list if `te`, `ter`, `term` is typed. Then I have to type `terminal`, or set all to the top.
> 
Don't think I will change the matching of top records to using contains because this will introduce unwanted side effects during search, sorry @HsunGong. The root cause of the problem you are experiencing is likely the search and scoring mechanism itself in Wox because from your screenshots you should be getting Excel rather than the other program. Additionally, if I am not wrong also the scoring after running the search method is not honoured in the result list (think I saw it in an issue upstream)

In other words user shouldn't be using top records functionality to get the result user wants to show at top while typing the search, Wox should be getting the right results to the top otherwise it's just not a good experience. Or at least use search history to identify the most preferred result.

**Comment by jjw24** [comment]
> Other cases is that, if I reindex the program, and all `top` disappeared.

Do you mean running Reload Data query command or reindexing via the Programs plugin settings?
